src/common/user-store/preferences-helpers.ts

```diff
@@ -222,6 +241,16 @@ const syncKubeconfigEntries: PreferenceDescription<KubeconfigSyncEntry[], Map<st
   },
 };
 
+const editorConfiguration: PreferenceDescription<EditorConfiguration, EditorConfiguration> = {
+  fromStore(val) {
+    return val || {lineNumbers: null, miniMap: null, tabSize: 4};
```

Comment:
I think we should store `Partial<EditorConfiguration>` but work with inside Lens always with `Required<EditorConfiguration>`. 

Having said that the `fromStore` function should populate the default values and `toStore` should remove the default values (even if some of the values have been set).

Reply:
Default values are pulled from export const defaultEditorConfig, store saves only changed values

Reply:
That is not what those two lines do though... The goal should be to produce `undefined` if all the values are the "default".